.github/workflows/update-oam-defs.yml

```
@@ -0,0 +1,108 @@
+name: Meshery Nginx Service Mesh Pattern Components Generator
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  check_nginxsm_version:
+    name: Check NGINX Service Mesh Version
+    if: github.repository == 'meshery/meshery-nginx-sm'
+    runs-on: ubuntu-latest
+    outputs:
+      NEW_NGINXSM_VERSION: ${{ steps.glrt.outputs.release }}
+      EXISTS: ${{ env.EXISTS }}
+    steps:
+      - name: Get latest release tag
+        id: glrt
+        uses: pozetroninc/github-action-get-latest-release@master
+        with:
+          repository: nginxinc/nginx-service-mesh
+          excludes: prerelease, draft
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Check if manifests for latest release exist
+        run: |
+          if [ -d templates/oam/workloads/workloads-${{ steps.glrt.outputs.release }} ]
+          then
+            echo "EXISTS=true" > $GITHUB_ENV
+          else
+            export "EXISTS=false" > $GITHUB_ENV
+          fi
+
+  generator:
+    name: Generate and push OAM definitions
+    needs: [check_nginxsm_version]
+    if: ${{ github.repository == 'meshery/meshery-nginx-sm' && !needs.check_nginxsm_version.outputs.EXISTS }}
+    env:
+      NEW_NGINXSM_VERSION: ${{ needs.check_nginxsm_version.outputs.NEW_NGINXSM_VERSION }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          ref: 'master'
+      - name: Get Nginx Service Mesh manifests
+        run: |
+          mkdir ./nginxsm-crds
+          cd ./nginxsm-crds
+          curl --remote-name-all `curl -H "Accept: application/vnd.github.VERSION.raw" https://api.github.com/repos/nginxinc/nginx-service-mesh/contents/helm-chart/crds\?ref\=$NEW_NGINXSM_VERSION | jq 'map(.download_url) | .[]' | tr -d '"'`
+          cd ..
+          touch nginxsm.yaml
+          awk 'FNR==1 && NR>1 { printf("\n%s\n\n","---") } 1' nginxsm-crds/*.yaml > ~/nginxsm.yaml
+      - name: Bulild jsonschema util
+        run: |
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          nvm install v14.15.3
+          nvm install-latest-npm
+          git clone https://github.com/layer5io/kubeopenapi-jsonschema util
+          cd util
+          npm i; npm i nexe -g
+          make linux
+      - name: Generate Workload definitions
+        run: |
+          dirPath=../templates/oam/workloads/workloads-$NEW_NGINXSM_VERSION
+          cd util
+          mkdir -p $dirPath
+          meshName=nginxsm
+          smpMeshName=NGINXSM
```

```suggestion
          smpMeshName=NGINX_SERVICE_MESH
```
SMP mesh name : )
https://github.com/service-mesh-performance/service-mesh-performance/blob/f5ffabafb5b4eed338123c4b10bb7942518f195a/spec/service_mesh.pb.go#L59